migration/idempotent/008-tracing-functions.sql

```diff
@@ -569,6 +569,28 @@ $func$
 LANGUAGE plpgsql;
 GRANT EXECUTE ON FUNCTION ps_trace.put_instrumentation_lib(text, text, bigint) TO prom_writer;
 
+-- Creates a temporary table (if it doesn't exist), suitable for tracing data ingestion.
+-- Supresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
+CREATE OR REPLACE FUNCTION _ps_trace.ensure_trace_ingest_temp_table(_temp_table_name text, _proto_table_name text)
+    RETURNS void
+    SECURITY DEFINER
+    VOLATILE
+    SET search_path = pg_catalog, pg_temp
+AS $func$
+BEGIN
+    IF _proto_table_name NOT IN ('span', 'link', 'event') THEN
+        RAISE EXCEPTION 'Invalid table prototype specified: %', _proto_table_name;
+    END IF;
+
+    SET LOCAL log_statement = 'none';
+
+    EXECUTE format($sql$CREATE TEMPORARY TABLE IF NOT EXISTS %I ON COMMIT DELETE ROWS AS TABLE _ps_trace.%I WITH NO DATA$sql$,
+                    _temp_table_name, _proto_table_name);
+END;
+$func$
+LANGUAGE plpgsql;
+GRANT EXECUTE ON FUNCTION _ps_trace.ensure_trace_ingest_temp_table(text, text) TO prom_writer;
```

Comment:
I think that generally it's good practice to revoke exec from PUBLIC on security definer functions.